server/api/store.go

```diff
@@ -231,16 +225,39 @@ func (h *storeHandler) SetState(w http.ResponseWriter, r *http.Request) {
 		h.rd.JSON(w, http.StatusBadRequest, "invalid state")
 		return
 	}
-
-	err := rc.SetStoreState(storeID, metapb.StoreState(state))
+	var err error
+	switch metapb.StoreState(state) {
+	case metapb.StoreState_Up:
+		err = rc.UpStore(storeID)
+	case metapb.StoreState_Offline:
+		err = rc.RemoveStore(storeID, false)
+	default:
+		h.rd.JSON(w, http.StatusBadRequest, "invalid state")
+		return
```

Comment:
I think we can add a case for `tombstone` and return the response that the tombstone is state is now allowed to be set.

Reply:
We already have the case in `TestStoreSetState`, the related test code
```
	// Invalid state.
	info = StoreInfo{}
	err = postJSON(testDialClient, url+"/state?state=Foo", nil)
	c.Assert(err, NotNil)
	err = postJSON(testDialClient, url+"/state?state=Tombstone", nil)
	c.Assert(err, NotNil)
```

Reply:
I mean the response for `tombstone` should be `not allowed` instead of `invalid state`